feat: use the v2 interface in `telemetry ui`

doudou opened the pull request:
On top of #400 

This uses Syskit to resolve orocos tasks, and uses the new protocol to show data. At least, it tremendously speeds up task resolution, and ensures that we see all the tasks Syskit sees (instead of only the tasks that are present on the main computer)

It removes the use / display of the task log, which does remove some information. This will be fixed in future PRs.

[sc-43835] [sc-43836] [sc-43837]

doudou committed:
feat: use the v2 interface in `telemetry ui`

This uses Syskit to resolve orocos tasks, and uses the new protocol
to show data. At least, it tremendously speeds up task resolution, and
ensures that we see all the tasks Syskit sees (instead of only the tasks
that are present on the main computer)

doudou committed:
fix: the task inspector needs async tasks

jhonasiv commented on lib/syskit/telemetry/ui/name_service.rb:
```diff
@@ -0,0 +1,68 @@
+                end
+
+                def include?(name)
+                    puts @registered_tasks.keys
```
This `puts` doesnt seem intentional

doudou replied:
It was already removed on my branch ... not sure what happened here.

jhonasiv commented on lib/syskit/interface/v2/protocol.rb:
Is this supposed to be in this PR? I seems like it should be done in another. I dont mind it staying since you are dealing with a bunch of "On top of" features, but I feel it needs more context, maybe a separate commit?

doudou replied:
This is a protocol modification that was done for the benefit of the runtime-view. I'll move it to a separate commit. Don't see the value of separating in another PR.

doudou replied:
Splitting done.

doudou committed:
feat: extend the protocol representation of Deployment

It adds information (orogen model names) needed to create
Orocos::TaskContext directly without any network access.

doudou committed:
fix: create syskit/interface/v2 to load v2-specific interface code
